src/main/java/org/broadinstitute/hellbender/engine/filters/UMIReadFilter.java

```diff
@@ -0,0 +1,42 @@
+package org.broadinstitute.hellbender.engine.filters;
+
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.hellbender.cmdline.ReadFilterArgumentDefinitions;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
+public class UMIReadFilter extends ReadFilter {
+    static final long serialVersionUID = 1L;
+    public static final String UMI_TAG = "RX";
+
+    @Argument(fullName= ReadFilterArgumentDefinitions.UMI_NAME, doc = "umi must have the format 'XXX-XXX'")
+    public String umi;
+
+    public UMIReadFilter() {};
+
+    @Override
+    public boolean test(GATKRead read) {
+        final String readUMI = read.getAttributeAsString(UMI_TAG);
+        final String umi1 = readUMI.split("-", 2)[0];
```

Comment:
Split once and save the array into a variable.

`Utils.split(String, char)` is substantially faster for non-regex splitting like this.

Reply:
Could you fix `UMI.getUMI()` while you're at it?  It does the same wasteful thing.    You might consider using that method as well to avoid reimplementing this.

Reply:
In fact, why not give UMI a string constructor, then make the argument a UMI, and just use UMI.equalsModuloOrder(). It would save reimplementing a bunch of stuff.